src/Polly.Core/Simmy/Outcomes/OutcomeStrategyOptions.TResult.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace Polly.Simmy.Outcomes;
+
+#pragma warning disable CS8618 // Required members are not initialized in constructor since this is a DTO, default value is null
+
+/// <summary>
+/// Represents the options for the Behavior chaos strategy.
+/// </summary>
+/// <typeparam name="TResult">The type of the outcome that was injected.</typeparam>
+public class OutcomeStrategyOptions<TResult> : MonkeyStrategyOptions
+{
+    /// <summary>
+    /// Gets or sets the delegate that's raised when the outcome is injected.
+    /// </summary>
+    /// <remarks>
+    /// Defaults to <see langword="null"/>.
+    /// </remarks>
+    public Func<OutcomeArguments<TResult, OnOutcomeInjectedArguments>, ValueTask>? OnOutcomeInjected { get; set; }
+
+    /// <summary>
+    /// Gets or sets the outcome generator to be injected for a given execution.
+    /// </summary>
+    /// <remarks>
+    /// Defaults to <see langword="null"/>. Either <see cref="Outcome"/> or this property is required.
+    /// When this property is <see langword="null"/> the <see cref="Outcome"/> is used.
+    /// </remarks>
+    public Func<OutcomeGeneratorArguments, ValueTask<Outcome<TResult>?>>? OutcomeGenerator { get; set; }
+
+    /// <summary>
+    /// Gets or sets the outcome to be injected for a given execution.
+    /// </summary>
+    /// <remarks>
+    /// Defaults to <see langword="null"/>. Either <see cref="OutcomeGenerator"/> or this property is required.
+    /// When this property is <see langword="null"/> the <see cref="OutcomeGenerator"/> is used.
+    /// </remarks>
+    public Outcome<TResult>? Outcome { get; set; }
```

Comment:
We might have a problem with disposable resources here. If outcome is disposable and is returned then it cannot be used for the next execution.


Maybe introduce somethign like?

``` csharp
public Func<TResult, ResultCloner, TResult>? ResultCloner { get; set; }
```

It will be optional delegate that when used it can clone this outcome (if it has result).

Can apply to this value only, the ones returned from `OutcomeGenerator` should be responsibility of the caller. 

Typical scenario is `HttpResponseMessage`

Reply:
Great point! How about just adding docs for the disposable outcomes scenarios and waiting for community feedback to eventually add the `ResultCloner` delegate? I'd avoid sacrificing the `Outcome` property since I think it's very convenient (plus I've never heard an issue about it) and Simmy users are already familiar with it, even I'd prefer adding the `ResultCloner` delegate as an optional parameter so that disposable outcomes can be handled properly rather than sacrificing the `Outcome` property for an scenario that might not always apply. Thoughts?

Reply:
I just know we will have this problem for `HttpResponseMessage` results, so for us the `Outcome` property will be a no-go. Anything that's disposable could cause issues down the line.

Could convenience extensions help here? (the ones you already have). Simple scenarios can be addressed by them.

But I am ok keeping this in this PR, and we can take another look at it in API Review. 

cc @clpun